transform/set_slice_mangler.go

```diff
@@ -0,0 +1,65 @@
+package transform
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var (
+	emptyStructType  = reflect.TypeOf(struct{}{})
+	emptyStructValue = reflect.ValueOf(struct{}{})
+)
+
+// SetSliceMangler conveniently maps from sets (e.g. map[string]struct{}) to
+// slices (e.g. []string)
+type SetSliceMangler struct {
+}
+
+// Mangle changes the type of the provided StructField from a map[T]struct{} to
+// []T.
+func (*SetSliceMangler) Mangle(sf reflect.StructField) ([]reflect.StructField, error) {
+	if sf.Type.Kind() == reflect.Map && sf.Type.Elem() == emptyStructType {
+		sf.Type = reflect.SliceOf(sf.Type.Key())
+	}
+
+	return []reflect.StructField{sf}, nil
+}
+
+// Unmangle turns []T back into a map[T]struct{}.  Naturally, deduplication will
+// occur.
+func (*SetSliceMangler) Unmangle(sf reflect.StructField, vs []FieldValueTuple) (reflect.Value, error) {
+	if sf.Type.Kind() == reflect.Map && sf.Type.Elem() == emptyStructType {
```

Comment:
Hmm, let's invert this if statement so we return early in the non-matching case and eliminate a level of indenting from most of the body.